app/models/solidus_content/entry.rb

```diff
@@ -6,8 +6,12 @@ class SolidusContent::Entry < ActiveRecord::Base
 
   scope :by_slug, ->(slug) { find_by!(slug: slug) }
 
-  def self.data_for(entry_type, slug)
-    SolidusContent::EntryType.find_by!(name: entry_type).entries.by_slug(slug).data
+  def self.data_for(entry_type_name, slug)
+    get(entry_type_name, slug).data
+  end
+
+  def self.get(entry_type_name, slug)
+    SolidusContent::EntryType.by_type_name(entry_type_name).entries.by_slug(slug)
```

Comment:
This is great! At this point we could even extract:

```rb
def self.by_type(type)
  unless type.is_a? SolidusContent::EntryType
    type = SolidusContent::EntryType.by_name(type)
  end
  where(entry_type: type)
end
```

and use `SolidusContent::Entry.by_type(:foo).by_slug(:bar)` everywhere, wdyt?

Reply:
I like that! Refactoring that now.